### ISSUE TITLE
`no-redeclare` and `no-shadow` not handling built-in globals.

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#3070]
## test.js

```js
/* eslint no-redeclare:2, no-shadow:2 */
var Object = 0;

function foo() {
    var Number = 0;
}
```

## Command & Actual

On master branch:

```
> git log -1
commit 9d1aeaf0b45e8576ad61b6bf03ff3384eec8745b
...
> node .\bin\eslint.js test.js --no-eslintrc
>
```

No errors.

## Expected

```
test.js
  2:5   error  Object is already defined                      no-redeclare
  5:9   error  Number is already declared in the upper scope  no-shadow

✖ 1 problems (1 errors, 0 warnings)
```

----

Currently, those pattern is warned by the `no-native-reassign` rule, but I think it is a work of the `no-redeclare` rule and the `no-shadow` rule.

The cause seems: 
- `no-redeclare` is checking a count of `variable.identifiers`, but built-in definitions don't have its identifier node.  Built-in variables have the `variable.writable` property, I can increase count with it.
- `no-shadow` seems to ignore variables which don't have `variable.identifiers`.

We should think carefully about `no-shadow` particularly. The browser environment includes variables which have general name such as `top`, `name`, `location`, `history`, `self`, `status`...
I'm sure `no-redeclare` should warn those (because `var top = 0;`-like declarations are ignored merely in a global scope of a browser), but I'm not sure `no-shadow` also should warn those.

![image](https://cloud.githubusercontent.com/assets/1937871/8763471/83e8374c-2dd8-11e5-8a0f-964390f6bb21.png)

I'm happy to work on this if accepted.

**Comment by nzakas** [comment]
I think if we add an option that allows people to opt-in to this behavior, then we can add it. Maybe `checkGlobals`?

**Comment by ilyavolodin** [comment]
If we add those options, do we need `no-native-reassign` anymore?

**Comment by nzakas** [comment]
Shadowing is about declaration, not assignment. So yes, we still need it.

**Comment by mysticatea** [comment]
Thanks. I will add the option.
